src/nisarqa/utils/stats_h5_writer/setup_writer.py

```diff
@@ -43,6 +43,47 @@ def setup_stats_h5_all_products(
     log.info(f"Input file's runconfig copied to {stats_file}")
 
 
+def setup_stats_h5_non_insar_products(
```

Comment:
Let's add a note to the docstring of `setup_stats_h5_all_products()` that indicates to users that this function should instead be used for non-InSAR products.

Alternatively, we could make `setup_stats_h5_all_products()` a private function and, in addition to `setup_stats_h5_non_insar_products()`, we could add a public `setup_stats_insar_products()` that is called for InSAR products. h/t @nemo794